kura/org.eclipse.kura.linux.position/src/main/java/org/eclipse/kura/linux/position/PositionProvider.java

```diff
@@ -0,0 +1,58 @@
+package org.eclipse.kura.linux.position;
+
+import java.time.LocalDateTime;
+
+import org.eclipse.kura.linux.position.GpsDevice.Listener;
+import org.eclipse.kura.position.NmeaPosition;
+import org.osgi.util.position.Position;
+
+public interface PositionProvider {
+
+    public void start();
+
+    public void stop();
+
+    public Position getPosition();
+
+    /**
```

Comment:
If this is a new interface, not clear the deprecations.

Reply:
to maintain the  compatibility with SerialDevicePositionProvider  than now implements Position Provider.

Reply:
Maybe we can split this in two internal interfaces? For example PositionProvider, which only allows to get the position and time, and NMEAPositionProvider that provides the NMEA-related informations.

Reply:
> Maybe we can split this in two internal interfaces? For example PositionProvider, which only allows to get the position and time, and NMEAPositionProvider that provides the NMEA-related informations.

I can do that, but also GpsdPositionProvider has to implement NMEAPositionProvider interface because PositionServiceImpl invoke the getNMEAPosition, unless i do a check on runtime class that I find an ugly solution.